diff --git a/src/common/datastorage.h b/src/common/datastorage.h
index 420ae08..f16088d 100644
--- a/src/common/datastorage.h
+++ b/src/common/datastorage.h
@@ -52,2 +52,3 @@
 #define XEX_OPT_ID_ENTRYPOINT      0x10100
+#define XEX_OPT_ID_IMAGE_BASE      0x10201
 #define XEX_OPT_ID_IMPORT_LIBS     0x103FF
diff --git a/src/setdata/optheaders.c b/src/setdata/optheaders.c
index 8efea93..826a021 100644
--- a/src/setdata/optheaders.c
+++ b/src/setdata/optheaders.c
@@ -324,5 +324,10 @@ int setOptHeaders(struct secInfoHeader *secInfoHeader, struct peData *peData, st
     optHeaderEntries->optHeaderEntry[currentHeader].dataOrOffset = secInfoHeader->baseAddr + peData->entryPoint;
     currentHeader++;
-
+
+    // Image base address (0x10201)
+    optHeaderEntries->optHeaderEntry[currentHeader].id = XEX_OPT_ID_IMAGE_BASE;
+    optHeaderEntries->optHeaderEntry[currentHeader].dataOrOffset = secInfoHeader->baseAddr;
+    currentHeader++;
+
     // Import libraries (0x103FF)
     if(importsPresent)
@@ -336,10 +341,5 @@ int setOptHeaders(struct secInfoHeader *secInfoHeader, struct peData *peData, st
         currentHeader++;
     }
-
-    // TLS info (0x20104)
-    optHeaderEntries->optHeaderEntry[currentHeader].id = XEX_OPT_ID_TLS_INFO;
-    setTLSInfo(&(optHeaders->tlsInfo));
-    currentHeader++;
-
+
     // System flags (0x30000)
     optHeaderEntries->optHeaderEntry[currentHeader].id = XEX_OPT_ID_SYS_FLAGS;
